test/e2e/framework/pod/wait.go

```diff
@@ -268,8 +268,7 @@ func WaitForPodCondition(c clientset.Interface, ns, podName, conditionDesc strin
 			}
 			return true, err
 		} else if err != nil {
-			// TODO(#109732): stop polling and return the error in this case.
-			e2elog.Logf("Error evaluating pod condition %s: %v", conditionDesc, err)
+			return false, fmt.Errorf("evaluating pod condition %s: %v", conditionDesc, err)
```

Comment:
I don't see why the err != nil can not be retryable based on the `condition()` signature

```
type podCondition func(pod *v1.Pod) (bool, error)
```

Reply:
/lgtm cancel

I think that this is also breaking 2 conformance tests https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/112503/pull-kubernetes-conformance-kind-ipv6-parallel/1570672755274158080

> Sep 16 07:40:01.591: INFO: Wanted all 1 pods to be running and ready, or succeeded. Result: false. Pods: [logs-generator]
> Sep 16 07:40:01.591: FAIL: Pod logs-generator was not ready

Reply:
So the reason why this simple change wasn't done is that we need to distinguish between retryable and permanent errors, and there's no mechanism for that yet.

I'll add one, stay tuned...

Reply:
👍